src/systems/system.C

```diff
@@ -2096,25 +2107,20 @@ Number System::point_value(unsigned int var, const Point & p, const Elem & e) co
 
   FEType fe_type = dof_map.variable_type(var);
 
-  // Build a FE so we can calculate u(p)
-  UniquePtr<FEBase> fe (FEBase::build(e.dim(), fe_type));
+  // Map the physical co-ordinates to the master co-ordinates using the inverse_map from fe_interface.h.
+  Point coor = FEInterface::inverse_map(e.dim(), fe_type, &e, p);
 
-  // Map the physical co-ordinates to the master co-ordinates using the inverse_map from fe_interface.h
-  // Build a vector of point co-ordinates to send to reinit
-  std::vector<Point> coor(1, FEInterface::inverse_map(e.dim(), fe_type, &e, p));
-
-  // Get the shape function values
-  const std::vector<std::vector<Real>> & phi = fe->get_phi();
+  // create this object to handle the correct
+  FEComputeData fe_data(this->get_equation_systems(), coor);
 
-  // Reinitialize the element and compute the shape function values at coor
-  fe->reinit (&e, &coor);
+  FEInterface::compute_data(e.dim(), fe_type, &e, fe_data);
```

Comment:
I'm about to be AFK, but remind me to check and make sure this works correctly with p refinement.  I had to change a lot of code when I added adaptive p refinement support, and I don't remember changing the FEComputeData stuff.  Maybe it was just so long ago that I've forgotten, but maybe I missed it...

Reply:
Okay - it looks like I added adaptive p refinement support to the FE codepath of compute_data(), but didn't feel safe touching the InfFE code path.  So there's definitely still work to be done there at some future date, but this change won't regress any existing p-refinement AMR codes, which is good enough for now.